checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInferenceLogic.java

```diff
@@ -43,104 +82,596 @@ public class MustCallInferenceLogic {
+      MethodInvocationNode mNode,
+      List<? extends VariableTree> paramsOfEnclosingMethod) {
+
+    List<Node> arguments = mcca.getArgumentsOfInvocation(mNode);
+    List<? extends VariableElement> parameters = mcca.getParametersOfInvocation(mNode);
+    if (parameters.isEmpty()) {
+      return;
+    }
+
+    for (int i = 0; i < arguments.size(); i++) {
+      Node arg = NodeUtils.removeCasts(arguments.get(i));
+      Element argElt = TreeUtils.elementFromTree(arg.getTree());
+      if (argElt == null && arg instanceof ArrayCreationNode) {
+        ArrayCreationNode arrayCreationNode = (ArrayCreationNode) arg;
+        List<Node> arrays = arrayCreationNode.getInitializers();
+        for (Node n : arrays) {
+          Element nElt = TreeUtils.elementFromTree(n.getTree());
+          if (nElt == null) {
+            continue;
+          }
+          if (nElt.getKind().isField()) {
+            isOwningField(n, mNode);
+          }
+        }
+      } else if (argElt != null && argElt.getKind().isField()) {
+        isOwningField(arg, mNode);
+      }
+
+      Set<ResourceAlias> argAliases = getResourceAliasOfArgument(obligations, arg);
+      for (int j = 0; j < paramsOfEnclosingMethod.size(); j++) {
+        VariableElement paramElt = TreeUtils.elementFromDeclaration(paramsOfEnclosingMethod.get(j));
+        for (ResourceAlias rl : argAliases) {
+          Element argAliasElt = rl.reference.getElement();
+          if (!argAliasElt.equals(paramElt)) {
+            continue;
+          }
+          JavaExpression target = JavaExpression.fromVariableTree(paramsOfEnclosingMethod.get(j));
+          if (mustCallObligationSatisfied(mNode, paramElt, target)) {
+            addOwningOnParams(i);
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Returns the set of resource aliases associated with the given argument node, by looking up the
+   * corresponding obligation in the set of obligations passed as an argument.
+   *
+   * @param obligations the set of obligations to search in
+   * @param arg the argument node whose corresponding resource aliases are to be returned
+   * @return the set of resource aliases associated with the given argument node, or an empty set if
+   *     the node does not
+   */
+  private Set<ResourceAlias> getResourceAliasOfArgument(Set<Obligation> obligations, Node arg) {
+    Node tempVar = mcca.getTempVarOrNode(arg);
+    if (!(tempVar instanceof LocalVariableNode)) {
+      return Collections.emptySet();
+    }
+
+    Obligation argumentObligation =
+        MustCallConsistencyAnalyzer.getObligationForVar(obligations, (LocalVariableNode) tempVar);
+    if (argumentObligation == null) {
+      return Collections.emptySet();
+    }
+    return argumentObligation.resourceAliases;
+  }
+
+  /**
+   * If the receiver of {@code mNode} is a candidate owning field and the method invocation
+   * satisfies the field's must-call obligation, then adds that field to the {@link
+   * #owningFieldToECM} set.
```

Comment:
This explanation seems incomplete, since this method does three things (checks arguments, the receiver, and indirect calls), but this comment only mentions one (the receiver).

Reply:
updated the doc to explain all these three checks.